Packs/jamf/Integrations/jamfV2/jamfV2.py

```diff
@@ -28,6 +29,73 @@
 
 
 class Client(BaseClient):
+    def __init__(self, base_url: str, verify: bool, username: str = "", password: str = "", proxy: bool = False):
+        super().__init__(base_url=base_url, auth=(username, password), verify=verify, proxy=proxy)
+
+        """ due to deprecating the basic auth option from the classical API versions 10.35 and up
+            the client will try to generate an auth token first, if it failed to do generate the token,
+            the client will use basic auth instead.
+        """
+        try:
+            self._token = self._get_token()
+            self._headers = {
+                'Authorization': f'Bearer {self._token}'
+            }
+            add_sensitive_log_strs(self._token)
+        except DemistoException as e:
+            demisto.log(str(e))
+            demisto.log("Couldn't create token will proceed using basic auth")
+
+    def _get_token(self) -> str:
+        """
+        Obtains token from integration context if available and still valid.
+        After expiration, new token are generated and stored in the integration context.
+        Returns:
+            str: token that will be added to authorization header.
+        """
+        integration_context = get_integration_context()
+        token = integration_context.get('token', '')
+        valid_until = integration_context.get('expires')
+
+        now_timestamp = arg_to_datetime('now').timestamp()  # type:ignore
+        # if there is a key and valid_until, and the current time is smaller than the valid until
+        # return the current token
+        if token and valid_until:
+            if now_timestamp < valid_until:
+                return token
+
+        # else generate a token and update the integration context accordingly
+        token = self._generate_token()
+
+        return token
+
+    def _generate_token(self) -> str:
+        """
+        Generates new token.
+        """
+        resp = self._http_request(method='POST', url_suffix='api/v1/auth/token', resp_type='json')
```

Comment:
Following this [comment](https://github.com/demisto/content/pull/19755#discussion_r913242820) - what do you get when the generation failed? it is not 403 or something of that sort?

Reply:
We get 403, Ive wrapped the token generation with ```try```. If the token generation has failed we will proceed using the basic auth.